docs-ui/components/sample.tsx

```diff
@@ -1,12 +1,79 @@
+import {createContext, ReactChild, useState} from 'react';
+import {ThemeProvider, useTheme} from '@emotion/react';
 import styled from '@emotion/styled';
 
+import {IconMoon} from 'app/icons';
 import space from 'app/styles/space';
+import {darkTheme, lightTheme, Theme} from 'app/utils/theme';
 
-const Sample = styled('div')`
+type ThemeName = 'dark' | 'light';
+
+type Props = {
+  children?: ReactChild;
+  showThemeSwitcher?: boolean;
+  noBorder?: boolean;
+};
+
+/** Expose the selected theme to children of <Sample /> */
+export const SampleThemeContext = createContext<ThemeName>('light');
+
+const Sample = ({children, showThemeSwitcher = false, noBorder = false}: Props) => {
+  const [theme, setTheme] = useState<ThemeName>('light');
+  let themeObject: Theme;
+
+  const toggleTheme = () => {
+    if (theme === 'light') {
+      setTheme('dark');
+    } else {
+      setTheme('light');
+    }
+  };
+
+  if (showThemeSwitcher) {
+    themeObject = theme === 'light' ? lightTheme : darkTheme;
+  } else {
+    themeObject = useTheme();
```

Comment:
You probably want `themeObject` to be in state, it probably works fine most of the time because you call `setTheme`, but this could lead to unwanted behavior because it's not in state.